### PR TITLE
Add resource-class to offer/lease

### PR DESCRIPTION
**Pull request by DanNiESh**

### CODE REVIEW

**Review by tzumainn — COMMENTED** [review]
Thinking about it - we may not want to actually store the resource_class field in the esi-leap database, as that's duplication of information. It may be better to fetch it from Ironic, in the same way that we fetch the node name from Ironic (in esi_leap/resource/objects/ironic.py). Take a look and let me know what you think!

**Review by tzumainn — CHANGES_REQUESTED** [review]
This looks pretty good! Two requested changes inline; let me know if you think those changes wouldn't be possible.

File: esi_leap/api/controllers/v1/lease.py [hunk 1]
```diff
@@ -117,10 +119,22 @@ def get_all(self, project_id=None, start_time=None, end_time=None,
...
+                        lease_with_added_info.get('resource_class')
+                    if lease_resource_class == resource_class:
+                        lease_collection.leases.append(
+                            Lease(**lease_with_added_info))
+            else:
+                lease_collection.leases = [
                     Lease(**utils.lease_get_dict_with_added_info(
-                        lease, project_list, node_list)))
+                          lease, project_list, node_list))
+                    for lease in leases]
```
[the remark this answers is not in the document]
Comment (DanNiESh) [reply esi_leap/api/controllers/v1/lease.py:137]: Ah, do you mean using one code block for both cases? Sure!

[the remark this answers is not in the document]
Comment (DanNiESh) [reply esi_leap/api/controllers/v1/lease.py]: Does it look good now?

[the remark this answers is not in the document]
Comment (tzumainn) [reply esi_leap/api/controllers/v1/lease.py]: Almost! I was thinking something more like (for offers):

````
offers_with_added_info = [Offer(**utils.offer_get_dict_with_added_info(o, project_list, node_list))
                for o in offers]
if resource_class:
  offer_collection.offers = [offer in offers_with_added_info if offer.get('resource_class') == resource_class]
else:
  offer_collection.offers = offers_with_added_info
````

[the remark this answers is not in the document]
Comment (DanNiESh) [reply esi_leap/api/controllers/v1/lease.py]: Ah, gotcha!

[the remark this answers is not in the document]
Comment (DanNiESh) [reply esi_leap/api/controllers/v1/lease.py]: Done!

**Pull request merged**